### DISCUSSION TITLE
Immich - addon size

### DISCUSSION THREAD

**Discussion by peeter22** [alexbelgium/hassio-addons#1074] [Q&A]
Hi.
I have notticed that my Immich addon has significantly grown in size. From 37 to 756MB to be exact.
![obrázok](https://github.com/alexbelgium/hassio-addons/assets/57701560/f57c112f-4164-4186-b750-27b7539fbe5e)

Is this normal with the latest updates, or have I done something incorrectly ?
I am bit limited in terms of HA space so I had to limit the number of locally stored backups due to this.

**Comment by alexbelgium** [comment]
Hi, could you look within this what takes the space? It could be your database (so useful) or training packages for the AI that I could exclude from the backups but that woudl still be required somehow

**Reply by peeter22** [reply]
Could you please point me to a direction, where could I find that ?

**Reply by alexbelgium** [reply]
For example you could download the backup file and explore it with 7zip. To do that you go in http://homeassistant.local/hassio/backups ; click on the backup, then select download

**Reply by peeter22** [reply]
So it is the AI. 
![obrázok](https://github.com/alexbelgium/hassio-addons/assets/57701560/cbd681e6-97c4-4e8c-a94d-8b35b59f93f6)
![obrázok](https://github.com/alexbelgium/hassio-addons/assets/57701560/0ffc3692-2442-4794-8a81-a8618ab09f54)

![obrázok](https://github.com/alexbelgium/hassio-addons/assets/57701560/6db65c19-fa47-48a0-ae98-84fad64e420f)

As my photo gallery is for now only 4000 photos and 20GiB in size, it looks like I need to go with the machine learning turned off.
In fact it has never worked too great.
Is it possible to delete the files created by machine learning ? If I just disable the funkction it looks like they remain (the backup size is te same)

**Comment by alexbelgium** [comment]
Well I can exclude the IA from the backup folders also ; but if it is recreated at each reinstall it could be suboptimal

There is 2 ways to delete the folder : with portainer, or with my system https://github.com/alexbelgium/hassio-addons/wiki/Add%E2%80%90ons-feature-:-customisation

For that, you create a file named something like /config/addons_autoscripts/immich.sh (exact path in addon log when starting), then put something like : 
```
#!/bin/bash
echo "Removing machine-learning"
rm -r /data/machine-learning
rm /config/addons_autoscripts/immich.sh
```
and restart

**Reply by peeter22** [reply]
Thanks a lot ! 
I was able to do it using the built in customisation script function.
For now I will have to keep machine learning disabled as my 3GB free space is not enough to keep the Backups. I will experiment with it once I upgrade my system storage.
